GitHub Actions: Windows builds don't abort if installing dependencies fails

Currently, our Windows builds on GH Actions show the following error when installing dependencies:

```
visualstudio2019community v16.10.4.0 [Approved]
visualstudio2019community package files install completed. Performing other installation steps.
Downloading channel manifest 
  from 'https://aka.ms/vs/16/release/channel'
ERROR: The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://aka.ms/vs/16/release/channel'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
The install of visualstudio2019community was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\visualstudio2019community\tools\ChocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 4/5 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Installed:
 - dotnetfx v4.8.0.20190930
 - chocolatey-visualstudio.extension v1.9.0
 - visualstudio-installer v2.0.1
 - chocolatey-dotnetfx.extension v1.0.1

Failures
 - visualstudio2019community (exited 404) - Error while running 'C:\ProgramData\chocolatey\lib\visualstudio2019community\tools\ChocolateyInstall.ps1'.
 See log for details.
```

However, the workflow continues until it actually tries to use VS and fails with the following error:

```
Exception: D:\a\icinga2\icinga2\windows-icinga2\icinga2\tools\win32\load-vsenv.ps1:45
Line |
  45 |    throw "Could not get Build environment script at locations: ${vcvar …
     |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Could not get Build environment script at locations: C:\Program Files (x86)\Microsoft Visual
     | Studio\2019\BuildTools\VC\Auxiliary\Build\vcvars64.bat C:\Program Files (x86)\Microsoft Visual
     | Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat

Error: Process completed with exit code 1.
```

The windows build scripts should fail early.

--- a/doc/win-dev.ps1
+++ b/doc/win-dev.ps1
@@ -51,6 +51,8 @@ function Install-Exe {
 	}
 
 	Start-Process -Wait -FilePath $ExeFile -ArgumentList @('/VERYSILENT', '/INSTALL', '/PASSIVE', '/NORESTART', "/DIR=${Dir}")
+	ThrowOnNativeFailure
+
 	Remove-Item -Recurse -Path $TempDir
 }
 
@@ -59,6 +61,7 @@ try {
 	Get-Command choco
 } catch {
 	Invoke-Expression (New-Object Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')
+	ThrowOnNativeFailure
 
 	$RegEnv = 'Registry::HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
 	$ChocoPath = ";$(Join-Path $Env:AllUsersProfile chocolatey\bin)"
@@ -69,19 +72,41 @@ try {
 
 
 choco install -y "visualstudio${VsVersion}community"
+ThrowOnNativeFailure
+
 choco install -y "visualstudio${VsVersion}-workload-netcoretools"
+ThrowOnNativeFailure
+
 choco install -y "visualstudio${VsVersion}-workload-vctools"
+ThrowOnNativeFailure
+
 choco install -y "visualstudio${VsVersion}-workload-manageddesktop"
+ThrowOnNativeFailure
+
 choco install -y "visualstudio${VsVersion}-workload-nativedesktop"
+ThrowOnNativeFailure
+
 choco install -y "visualstudio${VsVersion}-workload-universal"
+ThrowOnNativeFailure
+
 choco install -y "visualstudio${VsVersion}buildtools"
+ThrowOnNativeFailure
 
 
 choco install -y git
+ThrowOnNativeFailure
+
 choco install -y cmake
+ThrowOnNativeFailure
+
 choco install -y winflexbison3
+ThrowOnNativeFailure
+
 choco install -y windows-sdk-8.1
+ThrowOnNativeFailure
+
 choco install -y wixtoolset
+ThrowOnNativeFailure
 
 
 Install-Exe -Url "https://packages.icinga.com/windows/dependencies/boost_$($BoostVersion -join '_')-msvc-${MsvcVersion}-${Env:BITS}.exe" -Dir "C:\local\boost_$($BoostVersion -join '_')-Win${Env:BITS}"

